### DISCUSSION TITLE
Auth Password allowed characters

### DISCUSSION THREAD

**Discussion by l4rth** [supabase/supabase#5065] [Questions] [answered]
Hello,

I've made the observation that e.g. registering with a "+" in the password will lead to an successfull signup, but later the user can not sign in with his password.
I've tried to look for a list of characters in the documentation, however I only found such a list for the postgres password.

Which characters are allowed in a user password?
Where is this information documented?
Or do I have a problem with the usage of the supabase-js client?

Best regards

**Comment by awalias** [comment, marked as answer]
> registering with a "+" in the password will lead to an successfull signup, but later the user can not sign in with his password

this sounds like a bug, moving to our auth repo: https://github.com/supabase/gotrue/issues/354

and yes we should also add documentation for this

**Reply by l4rth** [reply]
Hi, thanks for the answer.

By your answer I got back to my code and reworked my signup-process and found a not so obvious bug in there, which lead to this error. So I assume it's not a bug in supabase. I also built a minimal example in the meantime and couldn't reproduce it there as well.

However I would really appreciate if this basic info would be documented

**Reply by ginacostanzo** [reply]
has documentation been added for this? i'm not able to find anything on password character rules (is there a minimum required length, must have a special character, etc?)

**Comment by Raigiku** [comment]
Hello friends, I found password validation in this [file](https://github.com/supabase/auth/blob/a6de8c94fc9a8cbf592fb38216419a0d638a172d/internal/api/password.go#L27).

**Reply by Raigiku** [reply]
And I believe the file that has the characters is [this](https://github.com/supabase/supabase/blob/9b9e334d00b8fa9b1ca3852f7685754604959ff0/apps/studio/components/interfaces/Auth/BasicAuthSettingsForm/BasicAuthSettingsForm.tsx#L38C7-L38C33) one
@ginacostanzo @l4rth  hope this is useful!
